ironfish-cli/src/commands/test-evm.ts

```diff
@@ -15,27 +17,49 @@ export class TestEvmCommand extends IronfishCommand {
     const node = await this.sdk.node()
     await node.openDB()
 
-    const blockchainDb = node.chain.blockchainDb
+    const evm = await IronfishEvm.create(node.chain.blockchainDb)
 
-    const evmState: EvmState = new EvmState(blockchainDb.db)
-    await evmState.init()
+    const senderKey = generateKey()
+    const receipientKey = generateKey()
 
-    Assert.isNotNull(evmState.manager)
+    const senderPrivateKey = Uint8Array.from(Buffer.from(senderKey.spendingKey, 'hex'))
+    const recipientPrivateKey = Uint8Array.from(Buffer.from(receipientKey.spendingKey, 'hex'))
 
-    const address = new Address(hexToBytes('0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b'))
+    const senderAddress = Address.fromPrivateKey(senderPrivateKey)
+    const recipientAddress = Address.fromPrivateKey(recipientPrivateKey)
 
-    let balance = (await evmState.manager.getAccount(address))?.balance ?? 0n
-    this.log(`Account at address ${address.toString()} has balance ${balance}`)
+    const senderAccount = new Account(BigInt(0), 500000n)
 
-    const account = new Account(BigInt(0), balance + 1000n)
+    await evm.stateManager.checkpoint()
+    await evm.stateManager.putAccount(senderAddress, senderAccount)
+    await evm.stateManager.commit()
 
-    await evmState.manager.checkpoint()
-    await evmState.manager.putAccount(address, account)
-    await evmState.manager.commit()
-    await evmState.manager.flush()
+    let senderBalance = (await evm.stateManager.getAccount(senderAddress))?.balance ?? 0n
+    this.log(
+      `Sender account at address ${senderAddress.toString()} has balance ${senderBalance}`,
+    )
 
-    balance = (await evmState.manager.getAccount(address))?.balance ?? 0n
-    this.log(`Account at address ${address.toString()} has balance ${balance}`)
+    const tx = new LegacyTransaction({
```

Comment:
What is the difference between LegacyTransaction and Transaction? I am guessing you chose the right one, just want to verify.

Reply:
There are a few different types of transactions based on EIPs: https://github.com/ethereumjs/ethereumjs-monorepo/blob/a8a89732efdbb2ceb57d80aa46afcfc294c84c0c/packages/tx/src/types.ts#L114

`LegacyTransaction` seemed like the simplest for testing 😅